readthedocs/templates/builds/build_detail.html

```diff
@@ -175,6 +175,17 @@
       {% endif %}
     {% endif %}
 
+    {% if build.finished and build.config.build.commands %}
```

Comment:
In general, I want to do away with these types of blocks in the build template. This is where I've described wanting code constructs to link error/warnings. But, that's a conversation we've been having for a while now.

But specifically to this block, this feels like an odd place to warn the user. The user has already enabled the feature for their builds for this to block to show. Why is it important to warn the user again at this point?

Perhaps the intention of this block was to instead say that "build.commands is in use for this build. If your build is hosed, you're on your own."

Reply:
To me, that doesn't feel like that's a case worth cluttering this UI for. There's any number of problems a user can introduce to their own configuration. Tuning this UI for edge case users, at the expense of normal usage, does negatively affect normal usage.

My main issue with this suggestion pattern in template logic is that multiple suggestions can show, regardless of build state, additional build errors, and existing notifications. And these suggestions are permanent, even after the user knows about it.

What about instead making this suggestion (but really all of these template suggestions eventually) notifications to the maintainer instead? This is triggered by first use of the feature or first project to use the feature, so we are providing the user with more obvious warning, which can be dismissed after reading, and we don't have to manage the notification with template logic.

Reply:
I agree with Eric here. I added this message on this page for that particular case and also to remind the users this feature is in beta _every_ time they take a look at the build page.

I prefer to not have a discussion about how to present notifications to users in this PR. I'd like to see a completely different approach than the one we currently have, but we are not there yet. I'm just following the pattern we already have for now --which I agree, it's not great.

> Tuning this UI for edge case users, at the expense of normal usage, does negatively affect normal usage.

I'm fine keeping the message, for now, we are launching a new beta feature to over-communicate our users first and remove/change it later when the feature is more established and people know more about it and its limitations. I don't think this will have a huge impact on many users.

Reply:
Strong agree on not trying to build a better notification system in this PR. I think this is warranted since it's a beta feature. This UX will be removed when we make it not beta, but for now the benefits outweigh the UX issues.

Reply:
If you want to do the work in bulk, that's okay, but I also don't feel there is anything stopping us from using a modern pattern here. We've used per-view notifications in the past and I feel we discussed a similar build notification not too long ago even. Are we expecting the pattern to not use a known pattern like notifications?

But we do need to prioritize this work either way. These blocks make templates harder to manage and pose design issues. I didn't even bother implementing this logic/blocks in the new templates build detail page.

Reply:
> That said, I think the template notification is good enough for now for this use case and I'm happy to work on a full replacement when working on the notifications project.

Fair enough. Lets keep up pressure to solve #9279 early then, and I'd love to prioritize the (much larger) #3399.

My main concern is getting stuck in the cyclical pattern of using the template logic because it's there, and then not making progress on these two issues because we've added too much debt for the problem to be solved easily.

Any little thing we can do to break this cycle is important.